### PR TITLE
[4925] Clicking Site Tools from the Launcher starts a different experience that feels disjointed

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4925

### CODE REVIEW

File: ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      hideSiteSwitcher={true}
+      activeToolId={activeToolId}
+      footerHtml={footerHtml}
+      openSidebar={openSidebar}
+      tools={tools}
+    >
+      {activeToolId ? (
+        <Suspencified>
+          <Widget {...tools.find((tool) => tool.url === activeToolId).widget} extraProps={{ embedded: false }} />
+        </Suspencified>
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx]: widget has a suspense. do we really need the suspenscified component?

File: ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            styles={{
+              root: {
+                height: '100%',
+                margin: 0
+              }
+            }}
+            title={<FormattedMessage id="siteTools.selectTool" defaultMessage="Please choose a tool from the left." />}
+            image={`${baseUrl}/static-assets/images/choose_option.svg`}
+          />
+        </Box>
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx]: isn't the empty state at the site tools app? it should be so we don't need multiple identical empty states from the different containers

File: ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx [hunk 3]
```diff
@@ -16,87 +16,83 @@
...
+  url: string;
+  widget: WidgetDescriptor;
 }
 
+export type SiteToolsAppProps = PropsWithChildren<{
+  site: string;
+  activeToolId: string;
+  footerHtml: string;
+  openSidebar: boolean;
+  width: number;
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx]: rename to sidebar width (all over)

File: ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx [hunk 4]
```diff
@@ -16,87 +16,83 @@
...
+  title: TranslationOrText;
+  icon: SystemIconDescriptor;
+  url: string;
+  widget: WidgetDescriptor;
 }
 
+export type SiteToolsAppProps = PropsWithChildren<{
+  site: string;
+  activeToolId: string;
+  footerHtml: string;
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx]: can be optional
don't show when embedded (just like the logo; hide the whole block)

File: ui/app/src/components/SiteToolsApp/UrlDrivenSiteTools.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            path={`/${tool.url}`}
+            render={() => {
+              return (
+                <Suspencified>
+                  <Widget {...tool.widget} extraProps={{ embedded: false }} />
+                </Suspencified>
+              );
+            }}
+          />
+        ))}
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/UrlDrivenSiteTools.tsx]: Looks like this is duplicate on the site tools app. We need a way to factor/abstract tool rendering to a single place.

File: ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/components/SiteToolsApp/EmbeddedSiteTools.tsx]: let discuss this

File: ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx [hunk 6]
```diff
@@ -129,76 +139,56 @@ export default function SiteToolsApp(props: SiteToolsAppProps) {
...
+              </section>
+              <EmptyState
+                styles={{
+                  root: {
+                    height: '100%',
+                    margin: 0
+                  }
+                }}
+                title="404"
+                subtitle={<FormattedMessage id={'siteTools.toolNotFound'} defaultMessage={'Tool not found'} />}
```
Comment (rart) [inline ui/app/src/components/SiteToolsApp/SiteToolsApp.tsx]:
```suggestion
                subtitle={<FormattedMessage id="siteTools.toolNotFound" defaultMessage="Tool not found" />}
```

**Review by rart — APPROVED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**